### ISSUE TITLE
Add a hover GestureRecognizer

### ISSUE THREAD

**Issue by jmagman** [flutter/flutter#33312]
## Use case
The hover gesture is an important one on devices with mouse support, particularly when  designing flutter for web.  A HoverGestureRecognizer would make it easier, for example, to add a recognizer to a TextSpan.

## Proposal
Create a GestureRecognizer HoverGestureRecognizer, similar to TapGestureRecognizer.

**Comment by DSenanayake** [comment]
Agreed!

**Comment by xxzj990** [comment]
I need it!!

**Comment by kenthinson** [comment]
Hover is already built in Listener()
However it seems to be broken on web. Not sure about other platforms as i did not test.

**Comment by DSenanayake** [comment]
@kenthinson yes, it's only working when you mouse down and enter.

**Comment by kenthinson** [comment]
@jmagman I have opened a issue to fix the bug. https://github.com/flutter/flutter/issues/33707
As it's not a feature request but rather a bug fix this one should be closed?

**Comment by jmagman** [comment]
I still think it would be nice to have a built-in hover GestureRecognizer that can be used to create a TextSpan.

**Comment by dkwingsmt** [comment]
@jmagman Can you illustrate what you want to achieve by this addition that can't be done with `MouseRegion.onHover` (which used to be `Listener.onHover` when this issue was opened)? Only with a clear application can we start to discuss the best way to design a new API.

**Comment by aliyazdi75** [comment]
@dkwingsmt `MouseRegion` takes `Widget` child, and can't convert `InlineSpan` TextSpan to `Widget`. Is that right?

**Comment by BaR488** [comment]
Why closed? Is HoverGestureRecognizer already exist or how handle PointerHoverEvent on TextSpan?

**Comment by dkwingsmt** [comment]
> Why closed? Is HoverGestureRecognizer already exist or how handle PointerHoverEvent on TextSpan?

All widgets now receive hover events. The gesture recognizer of TextSpan should receive hover events now (although I haven't specifically tested this lne).

**Comment by BaR488** [comment]
@dkwingsmt Thank you for your answer! But i not understand how handle hover event on TextSpan. Can you help me? How i can write custom gesture recognizer for PointerHoverEvent?

**Comment by gsx123** [comment]
I have same question about  how to handle hover event on TextSpan. 
Can any one show the demo code?

**Comment by kaboc** [comment]
I finally figured out how to change mouse cursor using `MouseRegion` on hover over `TextSpan`s with `GestureRecognizer`, thanks to [this video](https://www.youtube.com/watch?v=77GyeN5lbR0) by SuperDeclarative. He is demonstrating something different, but it greatly helped me get the idea.

![Hover over textspan](https://user-images.githubusercontent.com/20254485/110196045-36331680-7e85-11eb-97b3-459121e42434.gif)
https://dartpad.dev/39e105386893c6a7e7ab638dab375fdf?null_safety=true

This issue has been kept closed with only the hint of "All widgets now receive hover events. The gesture recognizer of TextSpan should receive hover events now" and questions after that were ignored. I thought it was really unhelpful. It took months before I found the solution.

Although I now know it is actually possible with `MouseRegion`, It'd be better if there was an easier way.
